src/rgw/rgw_admin.cc

```diff
@@ -3955,7 +3986,50 @@ int main(int argc, char **argv)
       cout << std::endl;
     }
     return 0;
-
+  case OPT_ROLE_CREATE:
+    {
+      if (role_name.empty()) {
+        cerr << "Empty role name" << std::endl;
+        return -EINVAL;
+      }
+      RGWRole role(g_ceph_context, store, role_name, path);
+      ret = role.create(true);
+      if (ret < 0) {
+        cerr <<"could not create role: " << err_msg << std::endl;
+        return -ret;
+      }
+      show_role_info(role, formatter);
+      return 0;
+    }
+  case OPT_ROLE_DELETE:
+    {
+      if (role_name.empty()) {
+        cerr << "Empty role name" << std::endl;
+        return -EINVAL;
+      }
+      RGWRole role(g_ceph_context, store, role_name);
+      ret = role.delete_obj();
+      if (ret < 0) {
+        cerr <<"could not delete role: " << err_msg << std::endl;
+        return -ret;
+      }
+      return 0;
+    }
+  case OPT_ROLE_LIST:
+    {
+      if (role_name.empty()) {
+        cerr << "Empty role name" << std::endl;
+        return -EINVAL;
+      }
+      RGWRole role(g_ceph_context, store, role_name);
+      ret = role.list();
+      if (ret < 0) {
+        cerr <<"could not list role: " << err_msg << std::endl;
+        return -ret;
+      }
+      show_role_info(role, formatter);
+      return 0;
+    }
```

Comment:
the 'role list' command feels more like a 'role get', doesn't it? will there ever be multiple entries to output here?